app/services/hyrax/work_resource_query_service.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+module Hyrax
+  # Responsible for retrieving information based on the given work.
+  #
+  # @see ProxyDepositRequest
+  # @see SolrDocument
+  # @see Hyrax::SolrService
+  # @note This was extracted from the ProxyDepositRequest, which was coordinating lots of effort. It was also an ActiveRecord object that required lots of Fedora/Solr interactions.
+  class WorkResourceQueryService
+    # @param [String] id - The id of the work
+    def initialize(id:)
+      @id = id
+    end
+    attr_reader :id
+
+    # @return [Boolean] if the work has been deleted
+    def deleted_work?
+      Hyrax.query_service.find_by(id: id)
+      false
+    rescue Valkyrie::Persistence::ObjectNotFoundError
+      true
+    end
+
+    def work
+      # Need to ensure it is a work?
+      resource = Hyrax.query_service.find_by(id: id)
```

Comment:
i'm still a little peeved (as i'm sure you must be too), to be doing all these checks. it feels like there should be one way to check if something is a registered "curation concern".

as an alternative thought, do we really need to check the registered concerns? i wonder if `resource.work?` wouldn't be good enough here.

Reply:
I was thinking the same.  Checking `resource.work?` feels sufficient to me.  It broadens the number of matching resources but I don't think the check to ensure it is a registered curation concern and not merely a work is really that important.

Looking ahead at other uses of `WorkRelation.new.find` I wonder if replacing them all with a `query_service.find_by` and check for `#work?` is acceptable.

Reply:
I went ahead and made the change to only test `#work?`.